app/utils/gallery/index.test.ts

```diff
@@ -0,0 +1,283 @@
+            expect(result.name).toBe(item.name);
+        });
+    });
+
+    describe('getShouldRender', () => {
+        it('should return true if index is within range of active index', () => {
+            expect(getShouldRender(5, 5)).toBe(true);
+            expect(getShouldRender(6, 5)).toBe(true);
+            expect(getShouldRender(9, 5)).toBe(false);
+        });
+    });
+
+    describe('measureItem', () => {
+        it('should measure and set shared values', () => {
+            const ref = jest.fn() as unknown as AnimatedRef<any>;
+            const sharedValues = {
+                x: {value: 0},
+                y: {value: 0},
+                width: {value: 0},
+                height: {value: 0},
+            } as GalleryManagerSharedValues;
+            measureItem(ref, sharedValues);
+            expect(sharedValues.x.value).toBe(100);
+            expect(sharedValues.y.value).toBe(200);
+        });
+
+        it('should measure and set shared values', () => {
+            const ref = jest.fn() as unknown as AnimatedRef<any>;
+            const sharedValues = {
+                x: {value: 0},
+                y: {value: 0},
+                width: {value: 0},
+                height: {value: 0},
+            } as GalleryManagerSharedValues;
+            measureItem(ref, sharedValues);
+            expect(sharedValues.x.value).toBe(100);
+            expect(sharedValues.y.value).toBe(200);
+        });
+
+        it('should handle measure exception and set shared values out of the viewport', () => {
+            const ref = jest.fn() as unknown as AnimatedRef<any>;
+            const sharedValues = {
+                x: {value: 0},
+                y: {value: 0},
+                width: {value: 0},
+                height: {value: 0},
+            } as GalleryManagerSharedValues;
+            (measure as jest.Mock).mockImplementationOnce(() => {
+                throw new Error('error');
+            });
+            measureItem(ref, sharedValues);
+            expect(sharedValues.x.value).toBe(999999);
+            expect(sharedValues.y.value).toBe(999999);
+        });
+    });
+
+    describe('openGalleryAtIndex', () => {
+        it('should open gallery and freeze other screens', () => {
+            const galleryIdentifier = 'gallery1';
+            const initialIndex = 0;
+            const items = [{id: '1', name: 'item1'}, {id: '2', name: 'item2'}] as GalleryItemType[];
+
+            openGalleryAtIndex(galleryIdentifier, initialIndex, items);
+            expect(Keyboard.dismiss).toHaveBeenCalled();
+            expect(Navigation.setDefaultOptions).toHaveBeenCalled();
+        });
+    });
+
+    describe('typedMemo', () => {
+        it('should memoize component', () => {
+            const component = jest.fn();
+            const memoizedComponent = typedMemo(component);
+
+            // @ts-expect-error type in typedef
+            expect(memoizedComponent.type).toBe(component);
```

Comment:
Would we know if `memoizedComponent` is a React.memo?  I wonder if there's an easier way to check vs creating a whole extra test file, like below:

I may be overthinking it, but I feel like below helps other devs to know that typedMemo is memoize-ing a component.

But, this doesn't give any gain in unit code coverage, just understanding of typedMemo func.

So 0/5.

```
// probably needs to be in a diff file since using renderer and React
// ex: typedMemo.test.tsx

import {render} from '@testing-library/react-native';
import React, {useEffect} from 'react';

import {typedMemo} from '.';


describe('typedMemo', () => {
    const checkRerender = jest.fn((width) => width);

    afterAll(() => {
        jest.clearAllMocks();
    });

    test('memoized component would not rerender if the prop didnt change', () => {
        const MemoizedComponent = typedMemo(SomeComponent);
        const {rerender} = render(<MemoizedComponent width={100}/>);

        rerender(<MemoizedComponent width={100}/>);

        expect(checkRerender).toHaveBeenCalledWith(100);
        expect(checkRerender).toHaveBeenCalledTimes(1);
    });

    test('memoized component would rerender if the prop did change', () => {
        const MemoizedComponent = typedMemo(SomeComponent);
        const {rerender} = render(<MemoizedComponent width={100}/>);

        rerender(<MemoizedComponent width={200}/>);

        expect(checkRerender).toHaveBeenCalledWith(100);
        expect(checkRerender).toHaveBeenCalledWith(200);
        expect(checkRerender).toHaveBeenCalledTimes(2);
    });

    test('control test, component not memoized, rerenders without prop change', () => {
        const {rerender} = render(<SomeComponent width={100}/>);

        rerender(<SomeComponent width={100}/>);

        expect(checkRerender).toHaveBeenCalledTimes(2);
    });
});

const SomeComponent = ({width}: { width: number}) => {
    useEffect(() => {
        checkRerender(width);
    });

    return null;
};
```